src/Algorithm/Greedy/GreedyAlgorithm.java

```diff
@@ -0,0 +1,52 @@
+package Algorithm.Greedy;
+
+import Loader.CVRPModel;
+import lombok.Data;
+
+@Data
+public class GreedyAlgorithm {
+    private Integer[] path;
+    private double cost;
+    private double[][] costMatrix;
+
+    private CVRPModel cvrpModel;
+
+    public GreedyAlgorithm(int startPoint, CVRPModel cvrpModel) {
+        this.path = new Integer[cvrpModel.getDatasetSize()];
+        this.path[0] = startPoint;
+        this.costMatrix = new double[cvrpModel.getDatasetSize()][cvrpModel.getDatasetSize()];
+        this.cvrpModel = cvrpModel;
+        this.cost = 0;
+    }
+
+    public void searchForSolution() {
+        deepCopyCostMatrix(cvrpModel.getAdjacencyMatrix());
+        double minCost = Double.MAX_VALUE;
+        int minCostIndex = -1;
+        double edgeCost;
+        for (int i = 0; i < this.path.length - 1; i++) {
+            for (int j = 0; j < this.path.length; j++) {
```

Comment:
Wydaje mi się, że tutaj indeksy nie są ok. Ogólnie jest tak, że skoro dataset size jest równy 101, to tak naprawdę mamy 100 konsumentów i 1 dystrybutornie. Zatem według mnie nie dość, że długość ścieżki powinna być dataset.size() - 1 (funkcja evaluateScore sama dolicza odwiedzanie dystrybutorni), to też w ścieżce nie powinno być odwiedzane to miejsce dystrybucji (czyli wg mnie nie powinno w niej być indeksu 0).